sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ContainerResource.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.resources;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/** Factory for {@link Resource} retrieving Container ID information. */
+public final class ContainerResource {
+
+  private static final Logger logger = Logger.getLogger(ContainerResource.class.getName());
+  private static final String UNIQUE_HOST_NAME_FILE_NAME = "/proc/self/cgroup";
+  private static final Pattern HEX_EXTRACTOR =
+      Pattern.compile("^([\\w]*?-)?([a-fA-F0-9]+)(\\.[\\w]*?)?$");
+
+  private final String cgroupFilePath;
+
+  // package private for testing purposes
+  ContainerResource(String cgroupFilePath) {
+    this.cgroupFilePath = cgroupFilePath;
+  }
+
+  /** Returns resource with container information. */
+  public static Resource get() {
+    ContainerResource factory = new ContainerResource(UNIQUE_HOST_NAME_FILE_NAME);
+    String containerId = factory.extractContainerId();
+
+    if (containerId == null) {
+      return Resource.empty();
+    }
+    return Resource.create(Attributes.of(ResourceAttributes.CONTAINER_ID, containerId));
+  }
+
+  /**
+   * Each line of cgroup file looks like "14:name=systemd:/docker/.../... A hex string is expected
+   * inside the last section separated by '/' Each segment of the '/' can contain metadata separated
+   * by either '.' or '-'
+   *
+   * <p>We see this with CRI-O "crio-abcdef1234567890ABCDEF.freetext", then use {@linkplain
+   * ContainerResource#HEX_EXTRACTOR} to extract the container hex id
+   *
+   * @return containerId
+   */
+  @Nullable
+  @SuppressWarnings("DefaultCharset")
+  public String extractContainerId() {
+    try (BufferedReader reader = new BufferedReader(new FileReader(cgroupFilePath))) {
```

Comment:
NIT: I'd prefer (but this may be a personal preference only) to first test file existence (File.exists(...)) instead of relying on exception (FileNotFound here).

Reply:
done